src/AST.h

```diff
@@ -17,6 +17,7 @@ class Location {
 	}
 
 	std::string fileName() const { return path ? path->generic_string() : ""; }
+	std::shared_ptr<fs::path> filePath() const { return path; }
```

Comment:
Did you intend to return a writable path here? At first glance it looks like this should really just return an `fs::path` by value.

Reply:
Oopps... that is an oversight on my part. The object return should off course not be writable.